### PR TITLE
NOTIF-150 Introduce behavior groups - Phase 1

### PR DESCRIPTION
**Pull request by gwenneg**
This is phase 1 of #170. It's meant to be non-breaking, without any effect on the production.

It can be used to test the behavior groups on CI/QA.

`notifications-frontend` does not have to be updated right away, this PR can be merged without breaking the UI.

### CODE REVIEW

**Comment by gwenneg** [comment]
Class diagram

**Comment by gwenneg** [comment]
⭐ New API operations

The 🔒 icon means that the operation involves an account ID check during the SQL query execution.

| | # |  Method | Path and description |
| - | - | - | - |
| | N1 | PUT | /internal/bundles/{id}/behaviorGroups/{behaviorGroupId}/default<br />Sets the default behavior group for a bundle. |
 | 🔒 | N2 | GET | /api/notifications/v1.0/notifications/eventTypes/affectedByRemovalOfBehaviorGroup/{behaviorGroupId}<br />Retrieves the event types (including their app) affected by the removal of a behavior group. |
 | 🔒 | N3 | PUT | /api/notifications/v1.0/notifications/eventTypes/{eventTypeId}/behaviorGroups/{behaviorGroupId}<br />Links a behavior group with an event type. |
 | 🔒 | N4 | DELETE | /api/notifications/v1.0/notifications/eventTypes/{eventTypeId}/behaviorGroups/{behaviorGroupId}<br />Unlinks a behavior group from an event type. |
 | 🔒 | N5 | GET | /api/notifications/v1.0/notifications/eventTypes/{eventTypeId}/behaviorGroups<br />Retrieves the behavior groups linked to an event type. |
 | 🔒 | N6 | PUT | /api/notifications/v1.0/notifications/eventTypes/{eventTypeId}/mute<br />Mutes an event type, removing all its link with behavior groups. |
 | 🔒 | N7 | POST | /api/notifications/v1.0/notifications/behaviorGroups<br />Creates a behavior group. |
 | 🔒 | N8 | PUT | /api/notifications/v1.0/notifications/behaviorGroups/{id}<br />Updates a behavior group. |
 | 🔒 | N9 | DELETE | /api/notifications/v1.0/notifications/behaviorGroups/{id}<br />Deletes a behavior group. |
 | 🔒 | N10 | PUT | /api/notifications/v1.0/notifications/behaviorGroups/{behaviorGroupId}/actions/{endpointId}<br />Adds an action (endpoint call) to a behavior group. |
 | 🔒 | N11 | DELETE | /api/notifications/v1.0/notifications/behaviorGroups/{behaviorGroupId}/actions/{endpointId}<br />Deletes an action (endpoint call) from a behavior group. |
 | 🔒 | N12 | GET | /api/notifications/v1.0/notifications/bundles/{bundleId}/behaviorGroups<br />Retrieves the behavior groups of a bundle. |
 | 🔒 | N13 | GET | /api/notifications/v1.0/notifications/bg/eventTypes/affectedByRemovalOfEndpoint/{endpointId}<br />Retrieve the event types affected by the removal of an integration. **Temporary path + runs the check with the new behavior groups classes** |

**Comment by gwenneg** [comment]
@theute @josejulio @pilhuhn This PR is ready for a review/merge.

We can talk about the model changes and the new APIs first, or we can do that after the merge. It doesn't really matter since this PR does not change any existing REST API or the Kafka messages processing.

File: src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java [hunk 1]
```diff
@@ -195,4 +276,79 @@ private String getAddress(Principal principal) {
...
+
+    @PUT
+    @Path("/behaviorGroups/{id}")
+    @Operation(summary = "Update a behavior group.")
+    @RolesAllowed(RbacIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
+    public Uni<Boolean> updateBehaviorGroup(@Context SecurityContext sec, @PathParam("id") UUID id, @NotNull @Valid BehaviorGroup behaviorGroup) {
+        return getAccountId(sec)
+                .onItem().transformToUni(accountId -> {
+                    behaviorGroup.setBundleId(id);
+                    return behaviorGroupResources.update(accountId, behaviorGroup);
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:306]: As is, would this allow anyone to take over other's `BehaviorGroup` ?

I think we should verify if `BehaviorGroup.accountId == accountId`.

**Review by josejulio — APPROVED** [review]
lgtm; I'm going to start playing with it :)

**Comment by gwenneg** [comment]
Thanks for the review @josejulio! Should we merge now or wait for your tests?

**Comment by josejulio** [comment]
I think we can merge and play with it on CI.

**Pull request merged**